core/src/scene/importer.cpp

```diff
@@ -0,0 +1,325 @@
+                    }
+
+                    return false;
+                });
+
+
+            if (m_sceneQueue.empty()) {
+                if (progressCounter == 0) {
+                    break;
+                }
+                continue;
+            }
+
+            path = m_sceneQueue.back();
+
+            m_sceneQueue.pop_back();
+
+            if (m_scenes.find(path) != m_scenes.end()) { continue; }
+        }
+
+        // TODO: generic handling of uri
+        std::regex r("^(http|https):/");
+        std::smatch match;
+        if (std::regex_search(path, match, r)) {
+            progressCounter++;
+            startUrlRequest(path,
+                    [&, p = path](std::vector<char>&& rawData) {
+
+                    if (!rawData.empty()) {
+                        std::unique_lock<std::mutex> lock(sceneMutex);
+                        processScene(p, std::string(rawData.data(), rawData.size()));
+                    }
+                    progressCounter--;
+                    m_condition.notify_all();
+            });
+        } else {
+            std::unique_lock<std::mutex> lock(sceneMutex);
+            processScene(path, getSceneString(path, resourceRoot));
+        }
+    }
+
+    auto root = importScenes(scenePath);
+
+    return root;
+}
+
+void Importer::processScene(const std::string &scenePath, const std::string &sceneString) {
+
+    std::string path = scenePath;
+    // Make sure all references from uber scene file are relative to itself, instead of being
+    // absolute paths (Example: when loading a file using command line args).
+    // TODO: Could be made better later.
+    if (m_scenes.size() == 0 && path[0] == '/') { path= getFilename(path); }
+
+    try {
+        auto root = YAML::Load(sceneString);
+        normalizeSceneImports(root, path);
+        normalizeSceneDataSources(root, scenePath);
+        normalizeSceneTextures(root, path);
+        auto imports = getScenesToImport(root);
+        m_scenes[path] = root;
+        for (const auto& import : imports) {
+            m_sceneQueue.push_back(import);
+            m_condition.notify_all();
+        }
+    }
+    catch (YAML::ParserException e) {
+        LOGE("Parsing scene config '%s'", e.what());
+    }
+}
+
+std::string Importer::getFilename(const std::string& filePath) {
+    std::string scenePath = filePath;
+    std::regex r("[^//]+$");
+    std::smatch match;
+    if (std::regex_search(scenePath, match, r)) { scenePath = match[0]; }
+    return scenePath;
+}
+
+std::string Importer::normalizePath(const std::string &path,
+                                         const std::string &parentPath) {
+
+    std::regex r("^(http|https):/");
+    std::smatch match;
+    if (path[0] == '/' || std::regex_search(path, match, r)) {
+        //absolute path or network url , return as it is
+        return path;
+    }
+
+    r = std::regex("[^//]+$");
+    return std::regex_replace(parentPath, r, path);
+}
```

Comment:
Could you add a comment what this replacement is supposed to do or an example case?